### PR TITLE
PLAT-40278 : Add background circle in Checkbox

### PR DESCRIPTION
**Pull request by changgilee**
Add background circle base on `18 GUI guidance in Enyo.

Enyo-DCO-1.1-Signed-off-by: Changgi Lee <changgi.lee@lge.com>

### CODE REVIEW

File: src/Checkbox/Checkbox.less [hunk 1]
```diff
@@ -2,22 +2,74 @@
...
+		}
+	}
+
 	.moon-icon {
-		visibility: hidden;
 		margin: 0;
 		padding: 0;
+		width: 30px;
+		height: 30px;
+		border-radius: 15px;
```
Comment (yeram) [inline src/Checkbox/Checkbox.less]: Based on `enact` we also need `line-height` as `30px` and `font-size` as `60px`.

File: src/Checkbox/Checkbox.less [hunk 2]
```diff
@@ -2,22 +2,73 @@
...
+			}
+		}
+	}
+
 	.moon-icon {
-		visibility: hidden;
 		margin: 0;
 		padding: 0;
+		width: 30px;
+		height: 30px;
```
Comment (Djspaceg) [inline src/Checkbox/Checkbox.less]: These would probably be better as a `@moon-checkbox-size` variable in the `variables.less` file

File: src/Checkbox/Checkbox.less [hunk 3]
```diff
@@ -2,22 +2,73 @@
...
+		line-height: 30px;
+		font-size: 60px;
```
Comment (Djspaceg) [inline src/Checkbox/Checkbox.less]: This line could become `font-size: (@moon-checkbox-size * 2);`

File: src/Checkbox/Checkbox.less [hunk 4]
```diff
@@ -2,22 +2,73 @@
...
+		border-radius: 15px;
```
Comment (Djspaceg) [inline src/Checkbox/Checkbox.less]: This line could become `font-size: (@moon-checkbox-size / 2);`

File: src/Checkbox/Checkbox.less [hunk 5]
```diff
@@ -2,22 +2,73 @@
...
+		color: transparent;
+		background-color: @moon-dark-gray;
```
Comment (Djspaceg) [inline src/Checkbox/Checkbox.less]: It's best practice to not refer to these variables outside of the `moonstone-dark/light.less` files, and instead assign the `@moon-dark-gray` to a more descriptive color variable name, like `@moon-checkbox-bg-color: @moon-dark-gray;`. This is simpler and easier to maintain over time and abstracts the color value from the component LESS files. Be sure to update both the light and dark color files.

File: src/Checkbox/Checkbox.less [hunk 6]
```diff
@@ -2,22 +2,73 @@
...
+			color: @moon-checkbox-toggle-switch-checked-color;
+		}
+		.moon-icon {
+			color: @moon-checkbox-toggle-switch-color;
+		}
+	}
+
+	.moon-checkbox:not([disabled]) {
+		.moon-icon {
+			background-color: rgba(255,255,255,0.41);
```
Comment (Djspaceg) [inline src/Checkbox/Checkbox.less]: This should be a color variable too. Also, do you think anyone can notice the difference between `0.41` and `0.4` alpha levels? Let's go ahead and round this value down to `0.4`: `rgba(255,255,255,0.4)`.
Alternatively, in the color file, you could use the LESS function to apply the alpha as well: `fade(@moon-white, 40%)`

**Review by Djspaceg — CHANGES_REQUESTED** [review]
Let's make a few organizational changes to this PR.

File: src/Checkbox/Checkbox.less [hunk 7]
```diff
@@ -2,22 +2,73 @@
...
+		}
+	}
+
+	.moon-checkbox:not([disabled]) {
+		.moon-icon {
+			background-color: fade(@moon-white, 40%);
+			color: transparent;
+		}
+		&[checked] .moon-icon {
+			color: @moon-white;
```
Comment (Djspaceg) [inline src/Checkbox/Checkbox.less]: Each of these `@moon-white` variables should be updated to be `@moon-spotlight-text-color`. This one, line 67, and 46.

**Pull request merged**